### PR TITLE
VACMS-13768: Adds other vet centers and restores status call

### LINKED ISSUE #13768: Force queue of VetCenter Facilities does not include all Outstations, mobiles or CAPS.
## Describe the defect
When we force queue VetCetner facilities for a bulk push to lighthouse, it is only doing VetCenters and is not doing outstations, mobiles or CAPs (should not be pushing caps now, but once CAPS are in Facility API we neeed to).
image

All the logic lives here modules/custom/va_gov_post_api/src/Form/VaGovFacilityForceQueueForm.php

## To Reproduce
1. visit https://prod.cms.va.gov/admin/config/va-gov-post-api/facility-force-queue
2. Notice that only Vet Cetnters are counted here.  (code was not updated when we split vet centers into 3 types)

## AC / Expected behavior
- [ ] I expect that when I visit the force queue page, there are options for Outstations, VC mobile ~, and CAPs~
- [ ] I expect that the Outstation and mobile options force queue nodes in all states except archived.
- [ ] ~I expect that the CAPs option  only force queuing PUBLISHED CAPs.  (node status = 1) Because they do not exist in the facility API until they are published in the CMS.~ CAPs need to wait for #13777, per @swirtSJW on [Slack](https://dsva.slack.com/archives/C0FQSS30V/p1687893324053059?thread_ts=1687886741.775019&cid=C0FQSS30V).
- [ ] ~I expect that the CAPs option clearly indicates that it is only force queuing PUBLISHED CAPs.~ 

## Screenshots
image

### PR DESCRIPTION
**Pull request by omahane**
## Description

Closes #13768 

## Testing done
Manually

### Vet Center (working once more)
Screen Shot 2023-06-27 at 4 44 52 PM

### Vet Center Outstation
Screen Shot 2023-06-27 at 4 45 15 PM

### Force push Vet Center Outstation
- [x] Run the[ force push](https://vacms-13768-force-queue-vet-ce-4joeu71a92bqu3efkvsviral9glkk6xm.demo.cms.va.gov/admin/config/va-gov-post-api/facility-force-queue)
- [x] Validate that "Helena Vet Center Outstation" (a published facility) is present in the queue
- [x] Validate that "Portsmouth Vet Center Outstation" (a facility in draft)  is present in the queue
- [x] Validate that "Navajo Vet Center Outstation" (an archived facility) is not one of those added to the queue.
### Force push Vet Center Mobile Vet Center
- [x] Run the [force push](https://vacms-13768-force-queue-vet-ce-4joeu71a92bqu3efkvsviral9glkk6xm.demo.cms.va.gov/admin/config/va-gov-post-api/facility-force-queue)
- [x] Validate that "Lexington Mobile Vet Center" (a published facility) is present in the queue
- [x] Validate that "Chinle Mobile Vet Center" (a facility in draft)  is present in the queue
- [x] Validate that "San Marcos Mobile Vet Center" (an archived facility) is not one of those added to the queue.
- [x] Compare total with number of archived
### Check that pre-existing pushes are fixed (or still working)
- [x] Check the [VAMC facility push](https://vacms-13768-force-queue-vet-ce-4joeu71a92bqu3efkvsviral9glkk6xm.demo.cms.va.gov/admin/config/va-gov-post-api/facility-force-queue).
- [x] Validate that all nodes are processed and some are queued for sync.
- [x] Check the [VAMC health services push](https://vacms-13768-force-queue-vet-ce-4joeu71a92bqu3efkvsviral9glkk6xm.demo.cms.va.gov/admin/config/va-gov-post-api/facility-force-queue).
- [x] Validate that all nodes are processed and some are queued for sync.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`

### CODE REVIEW

File: docroot/modules/custom/va_gov_post_api/src/Form/VaGovFacilityForceQueueForm.php [hunk 1]
```diff
@@ -145,8 +159,17 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
             if ($bundle === 'health_care_local_health_service') {
               $queued_count += _va_gov_post_api_add_facility_service_to_queue($node);
             }
+            // We don't push archived outstations or mobile vet centers.
+            elseif (
+              $bundle === 'vet_center_outstation' ||
+              $bundle === 'vet_center_mobile_vet_center'
+              ) {
+              if ($node->moderation_state->value !== 'archived') {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_post_api/src/Form/VaGovFacilityForceQueueForm.php:167]: What is the reason that this is the only one we are pre-checking for not archived?

Comment (omahane) [reply docroot/modules/custom/va_gov_post_api/src/Form/VaGovFacilityForceQueueForm.php:167]: Closing the loop, we should not force push any archived nodes through this bulk update, per Steve.

Comment (omahane) [reply docroot/modules/custom/va_gov_post_api/src/Form/VaGovFacilityForceQueueForm.php]: I have added an archived exclusion for all items, @swirtSJW

**Comment by swirtSJW** [comment]
Seeing one discrepancy which seems odd.  17 processed but only 16 queued.
image

**Comment by swirtSJW** [comment]
All the other counts match, except for VAMC... which makes sense because of Lovell facilities not sent
image

**Comment by swirtSJW** [comment]
Rebuilding this to get an empty queue so I can investigate the OS discrepancy.

**Comment by swirtSJW** [comment]
Christian found it.  Logan has no status yet so it does not get queued.

**Review by swirtSJW — APPROVED** [review]
Really nice work on this @omahane

**Comment by swirtSJW** [comment]
image

**Pull request merged**